src/app/components/shell/header/header.js

```diff
@@ -1,21 +1,88 @@
+import $ from '~/helpers/$';
 import BaseView from '~/helpers/backbone/view';
 import {on, props} from '~/helpers/backbone/decorators';
 import {template} from './header.hbs';
 
-function toggleClass(el, name) {
-    if (el.classList.contains(name)) {
-        el.classList.remove(name);
-    } else {
-        el.classList.add(name);
-    }
-}
-
 @props({
     el: '#header',
     template: template
 })
 export default class Header extends BaseView {
 
+    constructor() {
+        super(...arguments);
+
+        this.meta = {};
+
+        this.templateHelpers = {
+            collapsed: () => this.meta.collapsed,
+            fixed: () => this.meta.fixed,
+            transparent: () => this.meta.transparent
+        };
+    }
+
+    classList(action, ...args) {
+        var result = null;
+        let header = this.el.querySelector('.page-header');
+
+        if (header && typeof header.classList === 'object') {
+            result = header.classList[action](...args);
+        }
+
+        return result;
+    }
+
+    collapse() {
+        this.meta.collapsed = true;
+        this.classList('add', 'collapsed');
+        return this;
+    }
+
+    pin() {
+        this.meta.fixed = true;
+        this.classList('add', 'fixed');
+        return this;
+    }
+
+    transparent() {
+        this.meta.transparent = true;
+        this.classList('add', 'transparent');
+        return this;
+    }
+
+    reset() {
+        this.meta.collapsed = false;
+        this.meta.pinned = false;
+        this.meta.transparent = false;
+        this.classList('remove', 'collapsed');
+        this.classList('remove', 'fixed');
+        this.classList('remove', 'transparent');
+        return this;
+    }
+
+    isCollapsed() {
+        return !!this.classList('contains', 'collapsed');
+    }
+
+    isPinned() {
+        return !!this.classList('contains', 'fixed');
+    }
+
+    isTransparent() {
+        return !!this.classList('contains', 'transparent');
+    }
+
+    get height() {
+        let header = this.el.querySelector('.page-header');
```

Comment:
I think the original reasoning for a single `var` in a function scope was because `var` is hoisted to the top of the scope anyway, so it helps to prevent errors.  Since `let` is a bit more flexible and can be used in block scope, I don't think that's as relevant as a best practice any more.

In cases like that, I tend to defer to the most common practices in other languages, which use multiple declaration statements.  I don't think it has much impact on code clarity either way though, so I'm not opposed to leaving it up to the coder to decide what is best in a given situation (we could enforce the rule either way in ESLint, but I left it disabled at the moment at least to allow the choice).